packages/rollup-contracts/contracts/CanonicalTransactionChain.sol

```diff
@@ -59,23 +82,25 @@ contract CanonicalTransactionChain {
     uint numElementsInBatch = 1;
     bytes32 batchHeaderHash = keccak256(abi.encodePacked(
       timestamp,
-      true, // isL1ToL2Tx
+      isL1ToL2Tx,
       elementsMerkleRoot,
       numElementsInBatch,
       cumulativeNumElements // cumulativePrevElements
     ));
     batches.push(batchHeaderHash);
     cumulativeNumElements += numElementsInBatch;
-    l1ToL2Queue.dequeue();
   }
 
-  function appendTransactionBatch(bytes[] memory _txBatch, uint _timestamp) public {
+  function appendSequencerBatch(bytes[] memory _txBatch, uint _timestamp) public {
     require(authenticateAppend(msg.sender), "Message sender does not have permission to append a batch");
     require(_txBatch.length > 0, "Cannot submit an empty batch");
     require(_timestamp + forceInclusionPeriod > now, "Cannot submit a batch with a timestamp older than the sequencer inclusion period");
     require(_timestamp <= now, "Cannot submit a batch with a timestamp in the future");
     if(!l1ToL2Queue.isEmpty()) {
-      require(_timestamp <= l1ToL2Queue.peekTimestamp(), "Must process older queued batches first to enforce timestamp monotonicity");
+      require(_timestamp <= l1ToL2Queue.peekTimestamp(), "Must process older L1ToL2Queue batches first to enforce timestamp monotonicity");
+    }
+    if(!safetyQueue.isEmpty()) {
+      require(_timestamp <= safetyQueue.peekTimestamp(), "Must process older SafetyQueue batches first to enforce timestamp monotonicity");
```

Comment:
Can do same thing here where you remove the `if` and incorporate the if logic in the `require`